benchmarks/bench_feature_expansions.py

```diff
@@ -0,0 +1,70 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.sparse as sparse
+from sklearn.preprocessing import PolynomialFeatures
+from time import time
+
+degree = 2
+trials = 5
+num_rows = 100
+dimensionalities = [50, 100, 200, 400, 800]
+densities = [0.1, 0.4, 0.7, 1.0]
```

Comment:
I think it would be more interesting to explore densities such as  `[0.01, 0.1, 1]`. I think sparse representations only make sense below 0.1 density.